pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -256,7 +257,9 @@ func (jm *ControllerV2) getJobsToBeReconciled(cronJob *batchv1.CronJob) ([]*batc
 
 	for _, job := range jobList {
 		// If it has a ControllerRef, that's all that matters.
-		if controllerRef := metav1.GetControllerOf(job); controllerRef != nil && controllerRef.Name == cronJob.Name {
+		controllerRef := metav1.GetControllerOf(job)
+		// call resolveControllerRef to make sure the controllerRef is pointing to the right controller
+		if controllerRef != nil && jm.resolveControllerRef(job.Namespace, controllerRef) != nil {
```

Comment:
the name of the cronjob has to be taken into account. Please add tests that have multiple cronjobs in the store to be sure we are not filtering jobs of other cronjobs in this place

https://github.com/kubernetes/kubernetes/pull/120184#discussion_r1315780716

Reply:
Added a specific entry in test - `test getting jobs in namespace with a controller reference and if kind is present, should be CronJob` for this scenario. We are getting the job list based on the name of the controlllerRef.Name which would filter only jobs for that specific cron job.

Since this PR was closed - added a new one with the changes - https://github.com/kubernetes/kubernetes/pull/124222